internal/static/parse.go

```diff
@@ -0,0 +1,72 @@
+package static
+
+import (
+	"go/constant"
+	"go/types"
+
+	"github.com/dogmatiq/configkit"
+	"github.com/dogmatiq/configkit/internal/entity"
+	"github.com/dogmatiq/configkit/internal/typename/gotypes"
+	"golang.org/x/tools/go/ssa"
+)
+
+// parse parses the member of the package and populates it into
+// entity.Application.
+//
+// The second parameter is the type that actually implements
+// github.com/dogmatiq/dogma.Application interface.
+func parse(m ssa.Member, typ types.Type) *entity.Application {
+	app := &entity.Application{
+		TypeNameValue: gotypes.NameOf(typ),
+	}
+
+	fn := m.Package().Prog.LookupMethod(
+		typ,
+		m.Package().Pkg,
+		"Configure",
+	)
+
+	// Since `.Configure()` is the method, the first parameter is the receiver,
+	// so we need to extract the second paramenter as
+	// `dogma.ApplicationConfigurer`
```

Comment:
```suggestion
	// Since .Configure() is a method, the first parameter is the receiver, so
	// the second parameter is the `dogma.ApplicationConfigurer`.
```